app/v2/doc.go

```diff
@@ -0,0 +1,101 @@
+/*
+Package app provides an application framework that manages the live cycle of a running application.
+
+An app could be an HTTP server (or any kind of server), a worker or a simple program. Independently of the kind of the app, it always exposes an admin port at 9000 by default which serves metrics, debug information and kubernetes probes. It also is capable of graceful shutdown on receiving signals of terminating by itself.
```

Comment:
Nit: Maybe change "on receiving signals of terminating" -> "on receiving termination signals"?